operations/scheduler.go

```diff
@@ -694,6 +697,32 @@ func (s *Scheduler) executeOperationPreconditions(ctx context.Context, operation
 	return nil
 }
 
+func (s *Scheduler) FindConcurrentOperation(ctx context.Context, operation *types.Operation) (*types.Operation, error) {
```

Comment:
i noticed there's a `doesExistCascadeOperationForResource` function that does pretty much the same, consider reuse

Reply:
Refactored to a common function shared across this code and interceptors which require the same logic